swim/node.go

```diff
@@ -218,7 +218,7 @@ func (n *Node) Incarnation() int64 {
 
 func (n *Node) emit(event interface{}) {
 	for _, listener := range n.listeners {
-		go listener.HandleEvent(event)
```

Comment:
The only listeners that are registered are for stats and for hashring updates.

I think the hashring updates are better of being in sync, because we don't want the hashring to be out of sync with the membership.